### PR TITLE
[Chore] Add e2e cases for testing Tempo with AWS STS

### PR DESCRIPTION
**Pull request by IshwarKanse**
Add e2e tests for testing TempoStack and Tempo Monolithic support for AWS STS

```
$ chainsaw test --config .chainsaw-openshift.yaml tests/e2e-openshift-object-stores/
Version: v0.2.6
Loading config (.chainsaw-openshift.yaml)...
- Using test file: chainsaw-test
- TestDirs [tests/e2e-openshift-object-stores/]
- SkipDelete false
- FailFast false
- ReportFormat ''
- ReportName 'chainsaw-report'
- Namespace ''
- FullName false
- IncludeTestRegex ''
- ExcludeTestRegex ''
- ApplyTimeout 10s
- AssertTimeout 5m0s
- CleanupTimeout 5m0s
- DeleteTimeout 5m0s
- ErrorTimeout 5m0s
- ExecTimeout 5m0s
- DeletionPropagationPolicy Background
- NoCluster false
- PauseOnFailure false
Loading tests...
- aws-sts-monolithic (tests/e2e-openshift-object-stores/aws-sts-monolithic)
- aws-sts-tempostack (tests/e2e-openshift-object-stores/aws-sts-tempostack)
Loading values...
Running tests...
=== RUN   chainsaw
=== PAUSE chainsaw
=== CONT  chainsaw
=== RUN   chainsaw/aws-sts-monolithic
=== PAUSE chainsaw/aws-sts-monolithic
=== RUN   chainsaw/aws-sts-tempostack
=== PAUSE chainsaw/aws-sts-tempostack
=== CONT  chainsaw/aws-sts-monolithic
=== CONT  chainsaw/aws-sts-tempostack
=== NAME  chainsaw/aws-sts-monolithic
    | 06:05:44 | aws-sts-monolithic | @setup                                                     | CREATE    | OK    | v1/Namespace @ chainsaw-awssts-mono
    | 06:05:44 | aws-sts-monolithic | Create AWS S3 bucket, IAM policy and role required for STS | TRY       | RUN   |
=== NAME  chainsaw/aws-sts-tempostack
    | 06:05:44 | aws-sts-tempostack | @setup                                                     | CREATE    | OK    | v1/Namespace @ chainsaw-awssts-tempo
=== NAME  chainsaw/aws-sts-monolithic
    | 06:05:44 | aws-sts-monolithic | Create AWS S3 bucket, IAM policy and role required for STS | SCRIPT    | RUN   |
        === COMMAND
        /usr/bin/sh -c ./aws-sts-s3-create.sh tmmono chainsaw-awssts-mono
=== NAME  chainsaw/aws-sts-tempostack
    | 06:05:44 | aws-sts-tempostack | Create AWS S3 bucket, IAM policy and role required for STS | TRY       | RUN   |
    | 06:05:44 | aws-sts-tempostack | Create AWS S3 bucket, IAM policy and role required for STS | SCRIPT    | RUN   |
        === COMMAND
        /usr/bin/sh -c ./aws-sts-s3-create.sh tmstack chainsaw-awssts-tempo
=== NAME  chainsaw/aws-sts-monolithic
    | 06:05:49 | aws-sts-monolithic | Create AWS S3 bucket, IAM policy and role required for STS | SCRIPT    | LOG   |
        === STDOUT
        {
            "Location": "http://tracing-chainsaw-awssts-mono-ci-op-qlm139qm.s3.amazonaws.com/"
        }
        Creating IAM role 'tracing-chainsaw-awssts-mono-ci-op-qlm139qm'...
        Attaching role policy 'AmazonS3FullAccess' to role 'tracing-chainsaw-awssts-mono-ci-op-qlm139qm'...
        Role created and policy attached successfully!
        Create the secret to be used with Tempo
        secret/aws-sts created
    | 06:05:49 | aws-sts-monolithic | Create AWS S3 bucket, IAM policy and role required for STS | SCRIPT    | DONE  |
    | 06:05:49 | aws-sts-monolithic | Create AWS S3 bucket, IAM policy and role required for STS | ASSERT    | RUN   | v1/Secret @ chainsaw-awssts-mono/aws-sts
=== NAME  chainsaw/aws-sts-tempostack
    | 06:05:49 | aws-sts-tempostack | Create AWS S3 bucket, IAM policy and role required for STS | SCRIPT    | LOG   |
        === STDOUT
        {
            "Location": "http://tracing-chainsaw-awssts-tempo-ci-op-qlm139qm.s3.amazonaws.com/"
        }
        Creating IAM role 'tracing-chainsaw-awssts-tempo-ci-op-qlm139qm'...
        Attaching role policy 'AmazonS3FullAccess' to role 'tracing-chainsaw-awssts-tempo-ci-op-qlm139qm'...
        Role created and policy attached successfully!
        Create the secret to be used with Tempo
        secret/aws-sts created
    | 06:05:49 | aws-sts-tempostack | Create AWS S3 bucket, IAM policy and role required for STS | SCRIPT    | DONE  |
    | 06:05:49 | aws-sts-tempostack | Create AWS S3 bucket, IAM policy and role required for STS | ASSERT    | RUN   | v1/Secret @ chainsaw-awssts-tempo/aws-sts
=== NAME  chainsaw/aws-sts-monolithic
    | 06:05:49 | aws-sts-monolithic | Create AWS S3 bucket, IAM policy and role required for STS | ASSERT    | DONE  | v1/Secret @ chainsaw-awssts-mono/aws-sts
    | 06:05:49 | aws-sts-monolithic | Create AWS S3 bucket, IAM policy and role required for STS | TRY       | DONE  |
    | 06:05:49 | aws-sts-monolithic | Install Tempo Monolithic                                   | TRY       | RUN   |
    | 06:05:49 | aws-sts-monolithic | Install Tempo Monolithic                                   | APPLY     | RUN   | tempo.grafana.com/v1alpha1/TempoMonolithic @ chainsaw-awssts-mono/tmmono
=== NAME  chainsaw/aws-sts-tempostack
    | 06:05:49 | aws-sts-tempostack | Create AWS S3 bucket, IAM policy and role required for STS | ASSERT    | DONE  | v1/Secret @ chainsaw-awssts-tempo/aws-sts
    | 06:05:49 | aws-sts-tempostack | Create AWS S3 bucket, IAM policy and role required for STS | TRY       | DONE  |
    | 06:05:49 | aws-sts-tempostack | Install TempoStack                                         | TRY       | RUN   |
    | 06:05:49 | aws-sts-tempostack | Install TempoStack                                         | APPLY     | RUN   | tempo.grafana.com/v1alpha1/TempoStack @ chainsaw-awssts-tempo/tmstack
=== NAME  chainsaw/aws-sts-monolithic
    | 06:05:49 | aws-sts-monolithic | Install Tempo Monolithic                                   | CREATE    | OK    | tempo.grafana.com/v1alpha1/TempoMonolithic @ chainsaw-awssts-mono/tmmono
    | 06:05:49 | aws-sts-monolithic | Install Tempo Monolithic                                   | APPLY     | DONE  | tempo.grafana.com/v1alpha1/TempoMonolithic @ chainsaw-awssts-mono/tmmono
    | 06:05:49 | aws-sts-monolithic | Install Tempo Monolithic                                   | ASSERT    | RUN   | apps/v1/StatefulSet @ chainsaw-awssts-mono/tempo-tmmono
=== NAME  chainsaw/aws-sts-tempostack
    | 06:05:49 | aws-sts-tempostack | Install TempoStack                                         | CREATE    | OK    | tempo.grafana.com/v1alpha1/TempoStack @ chainsaw-awssts-tempo/tmstack
    | 06:05:49 | aws-sts-tempostack | Install TempoStack                                         | APPLY     | DONE  | tempo.grafana.com/v1alpha1/TempoStack @ chainsaw-awssts-tempo/tmstack
    | 06:05:49 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | RUN   | tempo.grafana.com/v1alpha1/TempoStack @ chainsaw-awssts-tempo/tmstack
    | 06:05:49 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | DONE  | tempo.grafana.com/v1alpha1/TempoStack @ chainsaw-awssts-tempo/tmstack
    | 06:05:49 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | RUN   | apps/v1/Deployment @ chainsaw-awssts-tempo/tempo-tmstack-query-frontend
    | 06:06:32 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | DONE  | apps/v1/Deployment @ chainsaw-awssts-tempo/tempo-tmstack-query-frontend
    | 06:06:32 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | RUN   | apps/v1/Deployment @ chainsaw-awssts-tempo/tempo-tmstack-distributor
    | 06:06:32 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | DONE  | apps/v1/Deployment @ chainsaw-awssts-tempo/tempo-tmstack-distributor
    | 06:06:32 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | RUN   | apps/v1/Deployment @ chainsaw-awssts-tempo/tempo-tmstack-querier
    | 06:06:32 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | DONE  | apps/v1/Deployment @ chainsaw-awssts-tempo/tempo-tmstack-querier
    | 06:06:32 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | RUN   | apps/v1/Deployment @ chainsaw-awssts-tempo/tempo-tmstack-compactor
=== NAME  chainsaw/aws-sts-monolithic
    | 06:06:45 | aws-sts-monolithic | Install Tempo Monolithic                                   | ASSERT    | DONE  | apps/v1/StatefulSet @ chainsaw-awssts-mono/tempo-tmmono
    | 06:06:45 | aws-sts-monolithic | Install Tempo Monolithic                                   | ASSERT    | RUN   | v1/Service @ chainsaw-awssts-mono/tempo-tmmono
    | 06:06:45 | aws-sts-monolithic | Install Tempo Monolithic                                   | ASSERT    | DONE  | v1/Service @ chainsaw-awssts-mono/tempo-tmmono
    | 06:06:45 | aws-sts-monolithic | Install Tempo Monolithic                                   | ASSERT    | RUN   | v1/Service @ chainsaw-awssts-mono/tempo-tmmono-jaegerui
    | 06:06:45 | aws-sts-monolithic | Install Tempo Monolithic                                   | ASSERT    | DONE  | v1/Service @ chainsaw-awssts-mono/tempo-tmmono-jaegerui
    | 06:06:45 | aws-sts-monolithic | Install Tempo Monolithic                                   | ASSERT    | RUN   | route.openshift.io/v1/Route @ chainsaw-awssts-mono/tempo-tmmono-jaegerui
    | 06:06:45 | aws-sts-monolithic | Install Tempo Monolithic                                   | ASSERT    | DONE  | route.openshift.io/v1/Route @ chainsaw-awssts-mono/tempo-tmmono-jaegerui
    | 06:06:45 | aws-sts-monolithic | Install Tempo Monolithic                                   | ASSERT    | RUN   | v1/ServiceAccount @ chainsaw-awssts-mono/tempo-tmmono
    | 06:06:45 | aws-sts-monolithic | Install Tempo Monolithic                                   | ASSERT    | DONE  | v1/ServiceAccount @ chainsaw-awssts-mono/tempo-tmmono
    | 06:06:45 | aws-sts-monolithic | Install Tempo Monolithic                                   | ASSERT    | RUN   | v1/Pod @ chainsaw-awssts-mono/tempo-tmmono-0
    | 06:06:45 | aws-sts-monolithic | Install Tempo Monolithic                                   | ASSERT    | DONE  | v1/Pod @ chainsaw-awssts-mono/tempo-tmmono-0
    | 06:06:45 | aws-sts-monolithic | Install Tempo Monolithic                                   | TRY       | DONE  |
    | 06:06:45 | aws-sts-monolithic | Wait for the TempoStack to be ready                        | TRY       | RUN   |
    | 06:06:45 | aws-sts-monolithic | Wait for the TempoStack to be ready                        | SCRIPT    | RUN   |
        === COMMAND
        /usr/bin/sh -c oc get --namespace chainsaw-awssts-mono tempomonolithic tmmono -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}' | grep True
    | 06:06:45 | aws-sts-monolithic | Wait for the TempoStack to be ready                        | SCRIPT    | LOG   |
        === STDOUT
        True
    | 06:06:45 | aws-sts-monolithic | Wait for the TempoStack to be ready                        | SCRIPT    | DONE  |
    | 06:06:45 | aws-sts-monolithic | Wait for the TempoStack to be ready                        | TRY       | DONE  |
    | 06:06:45 | aws-sts-monolithic | Generate traces                                            | TRY       | RUN   |
    | 06:06:45 | aws-sts-monolithic | Generate traces                                            | APPLY     | RUN   | batch/v1/Job @ chainsaw-awssts-mono/generate-traces
    | 06:06:45 | aws-sts-monolithic | Generate traces                                            | CREATE    | OK    | batch/v1/Job @ chainsaw-awssts-mono/generate-traces
    | 06:06:45 | aws-sts-monolithic | Generate traces                                            | APPLY     | DONE  | batch/v1/Job @ chainsaw-awssts-mono/generate-traces
    | 06:06:45 | aws-sts-monolithic | Generate traces                                            | ASSERT    | RUN   | batch/v1/Job @ chainsaw-awssts-mono/generate-traces
    | 06:06:49 | aws-sts-monolithic | Generate traces                                            | ASSERT    | DONE  | batch/v1/Job @ chainsaw-awssts-mono/generate-traces
    | 06:06:49 | aws-sts-monolithic | Generate traces                                            | TRY       | DONE  |
    | 06:06:49 | aws-sts-monolithic | Verify traces                                              | TRY       | RUN   |
    | 06:06:49 | aws-sts-monolithic | Verify traces                                              | APPLY     | RUN   | batch/v1/Job @ chainsaw-awssts-mono/verify-traces
    | 06:06:49 | aws-sts-monolithic | Verify traces                                              | CREATE    | OK    | batch/v1/Job @ chainsaw-awssts-mono/verify-traces
    | 06:06:49 | aws-sts-monolithic | Verify traces                                              | APPLY     | DONE  | batch/v1/Job @ chainsaw-awssts-mono/verify-traces
    | 06:06:49 | aws-sts-monolithic | Verify traces                                              | ASSERT    | RUN   | batch/v1/Job @ chainsaw-awssts-mono/verify-traces
    | 06:07:02 | aws-sts-monolithic | Verify traces                                              | ASSERT    | DONE  | batch/v1/Job @ chainsaw-awssts-mono/verify-traces
    | 06:07:02 | aws-sts-monolithic | Verify traces                                              | TRY       | DONE  |
    | 06:07:02 | aws-sts-monolithic | Verify traces                                              | CLEANUP   | RUN   |
    | 06:07:02 | aws-sts-monolithic | Verify traces                                              | DELETE    | RUN   | batch/v1/Job @ chainsaw-awssts-mono/verify-traces
    | 06:07:02 | aws-sts-monolithic | Verify traces                                              | DELETE    | OK    | batch/v1/Job @ chainsaw-awssts-mono/verify-traces
    | 06:07:02 | aws-sts-monolithic | Verify traces                                              | DELETE    | DONE  | batch/v1/Job @ chainsaw-awssts-mono/verify-traces
    | 06:07:02 | aws-sts-monolithic | Verify traces                                              | SCRIPT    | RUN   |
        === COMMAND
        /usr/bin/sh -c ./aws-sts-s3-delete.sh tmmono chainsaw-awssts-mono
    | 06:07:04 | aws-sts-monolithic | Verify traces                                              | SCRIPT    | LOG   |
        === STDOUT
        Remove IAM policy and delete role tracing-chainsaw-awssts-mono-ci-op-qlm139qm
        remove_bucket: tracing-chainsaw-awssts-mono-ci-op-qlm139qm
    | 06:07:04 | aws-sts-monolithic | Verify traces                                              | SCRIPT    | DONE  |
    | 06:07:04 | aws-sts-monolithic | Verify traces                                              | CLEANUP   | DONE  |
    | 06:07:04 | aws-sts-monolithic | Generate traces                                            | CLEANUP   | RUN   |
    | 06:07:04 | aws-sts-monolithic | Generate traces                                            | DELETE    | RUN   | batch/v1/Job @ chainsaw-awssts-mono/generate-traces
    | 06:07:04 | aws-sts-monolithic | Generate traces                                            | DELETE    | OK    | batch/v1/Job @ chainsaw-awssts-mono/generate-traces
    | 06:07:04 | aws-sts-monolithic | Generate traces                                            | DELETE    | DONE  | batch/v1/Job @ chainsaw-awssts-mono/generate-traces
    | 06:07:04 | aws-sts-monolithic | Generate traces                                            | CLEANUP   | DONE  |
    | 06:07:04 | aws-sts-monolithic | Install Tempo Monolithic                                   | CLEANUP   | RUN   |
    | 06:07:04 | aws-sts-monolithic | Install Tempo Monolithic                                   | DELETE    | RUN   | tempo.grafana.com/v1alpha1/TempoMonolithic @ chainsaw-awssts-mono/tmmono
    | 06:07:04 | aws-sts-monolithic | Install Tempo Monolithic                                   | DELETE    | OK    | tempo.grafana.com/v1alpha1/TempoMonolithic @ chainsaw-awssts-mono/tmmono
    | 06:07:04 | aws-sts-monolithic | Install Tempo Monolithic                                   | DELETE    | DONE  | tempo.grafana.com/v1alpha1/TempoMonolithic @ chainsaw-awssts-mono/tmmono
    | 06:07:04 | aws-sts-monolithic | Install Tempo Monolithic                                   | CLEANUP   | DONE  |
    | 06:07:04 | aws-sts-monolithic | @cleanup                                                   | DELETE    | RUN   | v1/Namespace @ chainsaw-awssts-mono
    | 06:07:04 | aws-sts-monolithic | @cleanup                                                   | DELETE    | OK    | v1/Namespace @ chainsaw-awssts-mono
    | 06:07:16 | aws-sts-monolithic | @cleanup                                                   | DELETE    | DONE  | v1/Namespace @ chainsaw-awssts-mono
=== NAME  chainsaw/aws-sts-tempostack
    | 06:07:23 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | DONE  | apps/v1/Deployment @ chainsaw-awssts-tempo/tempo-tmstack-compactor
    | 06:07:23 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | RUN   | apps/v1/StatefulSet @ chainsaw-awssts-tempo/tempo-tmstack-ingester
    | 06:07:23 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | DONE  | apps/v1/StatefulSet @ chainsaw-awssts-tempo/tempo-tmstack-ingester
    | 06:07:23 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | RUN   | v1/Service @ chainsaw-awssts-tempo/tempo-tmstack-compactor
    | 06:07:23 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | DONE  | v1/Service @ chainsaw-awssts-tempo/tempo-tmstack-compactor
    | 06:07:23 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | RUN   | v1/Service @ chainsaw-awssts-tempo/tempo-tmstack-distributor
    | 06:07:23 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | DONE  | v1/Service @ chainsaw-awssts-tempo/tempo-tmstack-distributor
    | 06:07:23 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | RUN   | v1/Service @ chainsaw-awssts-tempo/tempo-tmstack-gossip-ring
    | 06:07:23 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | DONE  | v1/Service @ chainsaw-awssts-tempo/tempo-tmstack-gossip-ring
    | 06:07:23 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | RUN   | v1/Service @ chainsaw-awssts-tempo/tempo-tmstack-ingester
    | 06:07:23 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | DONE  | v1/Service @ chainsaw-awssts-tempo/tempo-tmstack-ingester
    | 06:07:23 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | RUN   | v1/Service @ chainsaw-awssts-tempo/tempo-tmstack-querier
    | 06:07:23 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | DONE  | v1/Service @ chainsaw-awssts-tempo/tempo-tmstack-querier
    | 06:07:23 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | RUN   | v1/Service @ chainsaw-awssts-tempo/tempo-tmstack-query-frontend
    | 06:07:23 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | DONE  | v1/Service @ chainsaw-awssts-tempo/tempo-tmstack-query-frontend
    | 06:07:23 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | RUN   | v1/Service @ chainsaw-awssts-tempo/tempo-tmstack-query-frontend-discovery
    | 06:07:24 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | DONE  | v1/Service @ chainsaw-awssts-tempo/tempo-tmstack-query-frontend-discovery
    | 06:07:24 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | RUN   | route.openshift.io/v1/Route @ chainsaw-awssts-tempo/tempo-tmstack-query-frontend
    | 06:07:24 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | DONE  | route.openshift.io/v1/Route @ chainsaw-awssts-tempo/tempo-tmstack-query-frontend
    | 06:07:24 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | RUN   | v1/ServiceAccount @ chainsaw-awssts-tempo/tempo-tmstack
    | 06:07:24 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | DONE  | v1/ServiceAccount @ chainsaw-awssts-tempo/tempo-tmstack
    | 06:07:24 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | RUN   | v1/ServiceAccount @ chainsaw-awssts-tempo/tempo-tmstack-query-frontend
    | 06:07:24 | aws-sts-tempostack | Install TempoStack                                         | ASSERT    | DONE  | v1/ServiceAccount @ chainsaw-awssts-tempo/tempo-tmstack-query-frontend
    | 06:07:24 | aws-sts-tempostack | Install TempoStack                                         | TRY       | DONE  |
    | 06:07:24 | aws-sts-tempostack | Wait for the TempoStack to be ready                        | TRY       | RUN   |
    | 06:07:24 | aws-sts-tempostack | Wait for the TempoStack to be ready                        | SCRIPT    | RUN   |
        === COMMAND
        /usr/bin/sh -c oc get --namespace chainsaw-awssts-tempo tempo tmstack -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}' | grep True
    | 06:07:24 | aws-sts-tempostack | Wait for the TempoStack to be ready                        | SCRIPT    | LOG   |
        === STDOUT
        True
    | 06:07:24 | aws-sts-tempostack | Wait for the TempoStack to be ready                        | SCRIPT    | DONE  |
    | 06:07:24 | aws-sts-tempostack | Wait for the TempoStack to be ready                        | TRY       | DONE  |
    | 06:07:24 | aws-sts-tempostack | Generate traces                                            | TRY       | RUN   |
    | 06:07:24 | aws-sts-tempostack | Generate traces                                            | APPLY     | RUN   | batch/v1/Job @ chainsaw-awssts-tempo/generate-traces
    | 06:07:24 | aws-sts-tempostack | Generate traces                                            | CREATE    | OK    | batch/v1/Job @ chainsaw-awssts-tempo/generate-traces
    | 06:07:24 | aws-sts-tempostack | Generate traces                                            | APPLY     | DONE  | batch/v1/Job @ chainsaw-awssts-tempo/generate-traces
    | 06:07:24 | aws-sts-tempostack | Generate traces                                            | ASSERT    | RUN   | batch/v1/Job @ chainsaw-awssts-tempo/generate-traces
    | 06:07:27 | aws-sts-tempostack | Generate traces                                            | ASSERT    | DONE  | batch/v1/Job @ chainsaw-awssts-tempo/generate-traces
    | 06:07:27 | aws-sts-tempostack | Generate traces                                            | TRY       | DONE  |
    | 06:07:27 | aws-sts-tempostack | Verify traces                                              | TRY       | RUN   |
    | 06:07:27 | aws-sts-tempostack | Verify traces                                              | APPLY     | RUN   | batch/v1/Job @ chainsaw-awssts-tempo/verify-traces
    | 06:07:27 | aws-sts-tempostack | Verify traces                                              | CREATE    | OK    | batch/v1/Job @ chainsaw-awssts-tempo/verify-traces
    | 06:07:27 | aws-sts-tempostack | Verify traces                                              | APPLY     | DONE  | batch/v1/Job @ chainsaw-awssts-tempo/verify-traces
    | 06:07:27 | aws-sts-tempostack | Verify traces                                              | ASSERT    | RUN   | batch/v1/Job @ chainsaw-awssts-tempo/verify-traces
    | 06:07:30 | aws-sts-tempostack | Verify traces                                              | ASSERT    | DONE  | batch/v1/Job @ chainsaw-awssts-tempo/verify-traces
    | 06:07:30 | aws-sts-tempostack | Verify traces                                              | TRY       | DONE  |
    | 06:07:30 | aws-sts-tempostack | Verify traces                                              | CLEANUP   | RUN   |
    | 06:07:30 | aws-sts-tempostack | Verify traces                                              | DELETE    | RUN   | batch/v1/Job @ chainsaw-awssts-tempo/verify-traces
    | 06:07:30 | aws-sts-tempostack | Verify traces                                              | DELETE    | OK    | batch/v1/Job @ chainsaw-awssts-tempo/verify-traces
    | 06:07:30 | aws-sts-tempostack | Verify traces                                              | DELETE    | DONE  | batch/v1/Job @ chainsaw-awssts-tempo/verify-traces
    | 06:07:30 | aws-sts-tempostack | Verify traces                                              | SCRIPT    | RUN   |
        === COMMAND
        /usr/bin/sh -c ./aws-sts-s3-delete.sh tmstack chainsaw-awssts-tempo
    | 06:07:33 | aws-sts-tempostack | Verify traces                                              | SCRIPT    | LOG   |
        === STDOUT
        Remove IAM policy and delete role tracing-chainsaw-awssts-tempo-ci-op-qlm139qm
        remove_bucket: tracing-chainsaw-awssts-tempo-ci-op-qlm139qm
    | 06:07:33 | aws-sts-tempostack | Verify traces                                              | SCRIPT    | DONE  |
    | 06:07:33 | aws-sts-tempostack | Verify traces                                              | CLEANUP   | DONE  |
    | 06:07:33 | aws-sts-tempostack | Generate traces                                            | CLEANUP   | RUN   |
    | 06:07:33 | aws-sts-tempostack | Generate traces                                            | DELETE    | RUN   | batch/v1/Job @ chainsaw-awssts-tempo/generate-traces
    | 06:07:33 | aws-sts-tempostack | Generate traces                                            | DELETE    | OK    | batch/v1/Job @ chainsaw-awssts-tempo/generate-traces
    | 06:07:33 | aws-sts-tempostack | Generate traces                                            | DELETE    | DONE  | batch/v1/Job @ chainsaw-awssts-tempo/generate-traces
    | 06:07:33 | aws-sts-tempostack | Generate traces                                            | CLEANUP   | DONE  |
    | 06:07:33 | aws-sts-tempostack | Install TempoStack                                         | CLEANUP   | RUN   |
    | 06:07:33 | aws-sts-tempostack | Install TempoStack                                         | DELETE    | RUN   | tempo.grafana.com/v1alpha1/TempoStack @ chainsaw-awssts-tempo/tmstack
    | 06:07:33 | aws-sts-tempostack | Install TempoStack                                         | DELETE    | OK    | tempo.grafana.com/v1alpha1/TempoStack @ chainsaw-awssts-tempo/tmstack
    | 06:07:33 | aws-sts-tempostack | Install TempoStack                                         | DELETE    | DONE  | tempo.grafana.com/v1alpha1/TempoStack @ chainsaw-awssts-tempo/tmstack
    | 06:07:33 | aws-sts-tempostack | Install TempoStack                                         | CLEANUP   | DONE  |
    | 06:07:33 | aws-sts-tempostack | @cleanup                                                   | DELETE    | RUN   | v1/Namespace @ chainsaw-awssts-tempo
    | 06:07:33 | aws-sts-tempostack | @cleanup                                                   | DELETE    | OK    | v1/Namespace @ chainsaw-awssts-tempo
    | 06:07:45 | aws-sts-tempostack | @cleanup                                                   | DELETE    | DONE  | v1/Namespace @ chainsaw-awssts-tempo
--- PASS: chainsaw (0.00s)
    --- PASS: chainsaw/aws-sts-monolithic (92.33s)
    --- PASS: chainsaw/aws-sts-tempostack (120.91s)
PASS
Tests Summary...
- Passed  tests 2
- Failed  tests 0
- Skipped tests 0
Done.
```

### CODE REVIEW

**Comment by IshwarKanse** [comment]
cc @pavolloffay

**Review by pavolloffay — APPROVED** [review]

**Comment by IshwarKanse** [comment]
@andreasgerstmayr Need your blessing on this one.

**Review by frzifus — APPROVED** [review]

File: tests/e2e-openshift-object-stores/aws-sts-tempostack/aws-sts-s3-create.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::${aws_account_id}:oidc-provider/${oidc_provider}"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "${oidc_provider}:sub": [
+            "system:serviceaccount:${tempo_ns}:tempo-${tempo_name}",
+            "system:serviceaccount:${tempo_ns}:tempo-${tempo_name}-query-frontend"
```
Comment (andreasgerstmayr) [inline tests/e2e-openshift-object-stores/aws-sts-tempostack/aws-sts-s3-create.sh:40]: Is this line required? There is no `tempo-${tempo_name}-query-frontend` service account, neither in TempoStack nor in TempoMonolithic.

Comment (IshwarKanse) [reply tests/e2e-openshift-object-stores/aws-sts-tempostack/aws-sts-s3-create.sh:40]: @andreasgerstmayr The SA is created when we enable the ingress type as route in TempoStack instance. However its not created for the Monolithic, which uses a single SA.

Comment (IshwarKanse) [reply tests/e2e-openshift-object-stores/aws-sts-tempostack/aws-sts-s3-create.sh:40]: https://github.com/grafana/tempo-operator/pull/978#issuecomment-2224674870

Comment (andreasgerstmayr) [reply tests/e2e-openshift-object-stores/aws-sts-tempostack/aws-sts-s3-create.sh:40]: Oh, sorry I didn't realize we're creating a new SA here.

**Review by andreasgerstmayr — APPROVED** [review]

**Pull request merged**